### PR TITLE
[FIX] html_editor: properly convert self-closing tags

### PR DESCRIPTION
**Pull request by walidsahli**
**Problem**:
Previously, self-closing tags were converted after parsing with `DOMParser`. However, when parsing with the content type `"text/html"`, all self-closing tags are converted to open tags without corresponding closing tags.

**Solution**:
Convert self-closing tags before parsing to ensure proper handling.

**Steps to reproduce**:
1. Open email templates.
2. Open a template containing self-closing tags like `t` (e.g., Sales: Order Confirmation).
3. Observe that the template is incorrect, with all self-closing tags converted to open but not closed tags.

opw-4394833

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by walidsahli** [comment]
@odoo/rd-security can i have exception here ? i partially reverted a commit.

File: addons/html_editor/static/src/editor.js [hunk 1]
```diff
@@ -94,8 +94,7 @@ export class Editor {
         this.editable = editable;
         this.document = editable.ownerDocument;
         if (this.config.content) {
-            const content = fixInvalidHTML(this.config.content);
-            editable.innerHTML = content;
+            editable.innerHTML = this.config.content;
```
Comment (dmo-odoo) [inline addons/html_editor/static/src/editor.js:97]: I guess the new location where you call the function is more global than this one so you don't need this one anymore?

File: addons/html_editor/static/src/utils/sanitize.js [hunk 2]
```diff
@@ -58,7 +58,7 @@ export function initElementForEdition(element, options = {}) {
     }
 }
 
-export function fixInvalidHTML(content) {
+export function convertSelfClosingTagsToOpenClosed(content) {
```
Comment (dmo-odoo) [inline addons/html_editor/static/src/utils/sanitize.js:61]: Why did you rename the function? Self-closing tags are not valid in html (they are in xml), the current name is correct.

File: addons/html_editor/static/src/editor.js (hunk 1 above)
Comment (walidsahli) [reply addons/html_editor/static/src/editor.js:97]: Exactly, i dont need to fix the html here because it was fixed and parse at this stage.

**Comment by mart-e** [comment]
> i partially reverted a commit.

@walidsahli which one? Please add the reference to the commit. This way you can check why this line was added and what was the security justification at that time (and check it’s still valid).

File: addons/html_editor/static/src/utils/sanitize.js (hunk 2 above)
Comment (dmo-odoo) [reply addons/html_editor/static/src/utils/sanitize.js]: I see, but this function was specifically named that way so we could add other conversions to it in the future if needed without having to rename the function. In any case, even if I did agree with the renaming, it is an exported function in stable so we can't rename it anyway because other customers or partners could have imported it to use it in their own module. Honestly, I don't think they did, but they could, and we're not supposed to take the risk to break their code unless we have a very good reason to do so.

Comment (walidsahli) [reply addons/html_editor/static/src/utils/sanitize.js]: makes sense.

**Comment by walidsahli** [comment]
@mart-e this was the [commit](https://github.com/odoo/odoo/commit/26b922ef5cad42da7e188195e919e54878d472fc) which was reverted (partially)

**Comment by dmo-odoo** [comment]
@walidsahli To be fair it's not reverted per se, it's just that the line it added is moved higher up in the architecture such that it impacts more places, thus removing the need for the line in the original location it was added in the first place.

**Comment by walidsahli** [comment]
@dmo-odoo makes sense yea, it is just a reverting in term of diff but as you said it was moved up in the architecture.

**Comment by mart-e** [comment]
@walidsahli thanks but this doesn’t explain why your code is safe. The history can be helpfull but we still needs a justification.
Where does `this.config.content` comes from and how can I be sure it won’t evaluate untrusted html code? 
In this case, it was explained at https://github.com/odoo/odoo/pull/149028#issuecomment-2200240486 and it is assumed to be trusted as coming from a sanitized html field.

@robodoo override=ci/security

**Comment by dmo-odoo** [comment]
@robodoo r+
